raiden/messages/withdraw.py

```diff
@@ -130,6 +130,9 @@ def from_event(cls, event):
 
     def _data_to_sign(self) -> bytes:
         return pack_data(
+            (self.cmdid.value, "uint8"),
```

Comment:
If the goal is to differentiate the signature shouldn't the `cmdid` and perhaps also `nonce` be added to all other withdraw message types? Why only have it in `WithdrawExpired`? What is special about this message? Is it the only one that does not go in the contracts?

It looks a bit odd to only have it here.

Reply:
The special thing about the `WithdrawExpired` signature is indeed that it does not go to the contracts. Since we have different conventions for signatures that are used onchain compared to other ones, it is a consequence that this makes the messages inconsistent in this way. The difference exists on purpose to prevent signatures being reused onchain which are not created with that intention.

Apart from that, I do think that not including all fields in the signature in some way is clearly bad. Fixing that would result in larger changes than this PR. This is tracked in https://github.com/raiden-network/raiden-contracts/issues/1110 which has been pushed after the next release when the feature freeze started.

Reply:
Okay so I expected this but wanted to make sure. Thanks for pointing me to the further work issue.